.github/workflows/static_code_analysis_workflow.yml

```
@@ -0,0 +1,59 @@
+name: Reusable workflow for static code analysis
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        required: true
+        type: string
+      cargo-features-flag:
+        required: true
+        type: string
+
+jobs:
+  static-code-analysis-rust:
+    timeout-minutes: 40
+    runs-on: ${{ inputs.os }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # ratchet:actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@48d8f15b2aaa3d255ca5af3eba4870f807ce6b3c # ratchet:tj-actions/changed-files@v45
+
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@888c2e1ea69ab0d4330cbf0af1ecc7b68f368cc1 # ratchet:dtolnay/rust-toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt, clippy
+
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@bafd42bd91680affe190f697cac6ef3ad958ecba # ratchet:mozilla-actions/sccache-action@v0.0.9
+      - name: Set C/C++ caching env vars
+        run: |
+          echo "SCCACHE_GHA_ENABLED=true" >> $GITHUB_ENV
+          echo "RUSTC_WRAPPER=sccache" >> $GITHUB_ENV
+
+      - name: Run cargo fmt
+        run: cargo fmt --all -- --check
+
+      - name: Run cargo clippy
+        run: cargo clippy ${{ inputs.cargo-features-flag }} -- -D warnings
+
+      - name: Run code examples in documentation
+        run: cargo test --workspace --doc ${{ inputs.cargo-features-flag }}
+
+      - name: Build documentation
+        run: cargo doc ${{ inputs.cargo-features-flag }}
```

We can potentially move these steps into an own documentation job (running only on Ubuntu) to optimize the overall timing of the PR-Pipeline. Both jobs take up to 10 mins each and are blocking the rest of the Pipeline. 
It make more sense to have here fast checks that fail early. Everything that takes longer needs to be executed later.